tests/lib/ocpp/v201/test_smart_charging_handler.cpp

```diff
@@ -675,14 +683,91 @@ TEST_F(ChargepointTestFixtureV201, K01FR06_ExisitingProfileHasValidPeriodIncomin
 
+    auto periods = create_charging_schedule_periods(0);
     auto profile = create_charging_profile(
         DEFAULT_PROFILE_ID + 1, ChargingProfilePurposeEnum::TxDefaultProfile,
-        create_charge_schedule(ChargingRateUnitEnum::A, {}, {}), uuid(), ChargingProfileKindEnum::Absolute,
-        DEFAULT_STACK_LEVEL, ocpp::DateTime("2024-01-15T13:00:00"), ocpp::DateTime("2024-02-01T13:00:00"));
+        create_charge_schedule(ChargingRateUnitEnum::A, periods, ocpp::DateTime("2024-01-17T17:00:00")), uuid(),
+        ChargingProfileKindEnum::Absolute, DEFAULT_STACK_LEVEL, ocpp::DateTime("2024-01-15T13:00:00"),
+        ocpp::DateTime("2024-02-01T13:00:00"));
 
-    auto sut = handler.validate_tx_default_profile(profile, DEFAULT_EVSE_ID);
+    auto sut = handler.validate_profile(profile, DEFAULT_EVSE_ID);
 
     EXPECT_THAT(sut, testing::Eq(ProfileValidationResultEnum::DuplicateProfileValidityPeriod));
 }
 
+TEST_F(ChargepointTestFixtureV201, K01_ValidateProfile_IfEvseDoesNotExist_ThenProfileIsInvalid) {
+    auto profile = create_charging_profile(DEFAULT_PROFILE_ID, ChargingProfilePurposeEnum::TxProfile,
+                                           create_charge_schedule(ChargingRateUnitEnum::A), uuid());
+
+    auto sut = handler.validate_profile(profile, DEFAULT_EVSE_ID + 1);
+
+    EXPECT_THAT(sut, testing::Eq(ProfileValidationResultEnum::EvseDoesNotExist));
+}
+
+TEST_F(ChargepointTestFixtureV201, K01_ValidateProfile_IfScheduleIsInvalid_ThenProfileIsInvalid) {
+    create_evse_with_id(DEFAULT_EVSE_ID);
+    auto periods = create_charging_schedule_periods(0);
+    auto profile = create_charging_profile(
+        DEFAULT_PROFILE_ID, ChargingProfilePurposeEnum::TxProfile,
+        create_charge_schedule(ChargingRateUnitEnum::A, periods, ocpp::DateTime("2024-01-17T17:00:00")), uuid(),
+        ChargingProfileKindEnum::Relative, 1);
+
+    auto sut = handler.validate_profile(profile, DEFAULT_EVSE_ID);
+
+    EXPECT_THAT(sut, testing::Eq(ProfileValidationResultEnum::ChargingProfileExtraneousStartSchedule));
+}
+
+TEST_F(ChargepointTestFixtureV201, K01_ValidateProfile_IfChargeStationMaxProfileIsInvalid_ThenProfileIsInvalid) {
+    create_evse_with_id(DEFAULT_EVSE_ID);
+    std::string same_transaction_id = uuid();
+    open_evse_transaction(DEFAULT_EVSE_ID, same_transaction_id);
+    auto periods = create_charging_schedule_periods({0, 1, 2});
+    auto profile = create_charging_profile(
+        DEFAULT_PROFILE_ID, ChargingProfilePurposeEnum::ChargingStationMaxProfile,
+        create_charge_schedule(ChargingRateUnitEnum::A, periods, ocpp::DateTime("2024-01-17T17:00:00")), uuid());
+
+    auto sut = handler.validate_profile(profile, DEFAULT_EVSE_ID);
+
+    EXPECT_THAT(sut, testing::Eq(ProfileValidationResultEnum::ChargingStationMaxProfileEvseIdGreaterThanZero));
+}
+
+TEST_F(ChargepointTestFixtureV201, K01_ValidateProfile_IfTxDefaultProfile_K01FR52_IsInvalid_ThenProfileIsInvalid) {
```

Comment:
If feel off to me that the test name's are much less specific that the EXPECT_THAT/ What about something like `K01_ValidateProfile_IfDuplicateTxDefaultProfileFound_ThenProfileIsInvalid` or something that gives more context? These names are all very generic.

Reply:
The problem here is "IfDuplicateTxDefaultProfileFound" applies to both K01.FR.52 and K01.FR.53, so I don't know of a way to specify what *exactly* we're checking for here within these test names.

Reply:
One is DuplicateTxDefaultProfileFoundOnSpecificEVSE and one is DuplicateSystemWideTxDefaultProfileFound maybe?

Reply:
Updated :)